### ISSUE TITLE
Publish to Tibber 

### ISSUE THREAD

**Issue by ggrote** [jsphuebner/esp-egycounter#2]
Hi,

leider kann ich dich anders nicht erreichen, daher als Github Ticket:
Ich versuche auch mit meinem eigenen Lesekopf Daten an Tibber zu senden und bin dabei auf dein Projekt gestoßen.
Hast du mal versucht wie es sich ohne das Senden der Metrics und Events verhält? Also wenn man nur die Daten des Zählers sendet?
Oder mag Tibber das dann auch nicht?

Beste Grüße

**Comment by jsphuebner** [comment]
Ja, es hat anfangs die "May the force be with you" message gefehlt, daraufhin hat Tibber irgendwann keine Daten mehr angenommen. Ich vermute ähnlich verhielte es sich bei fehlenden Metrics, habe ich aber nicht probiert

**Comment by ggrote** [comment]
Ah okay, ich dachte zuerst das wäre ein Scherz von dir mit der Message. :)
Hast du einen Screenshot wie es in der Tibber App aussehen muss? Also woran merke ich, dass die Werte angenommen werden?

**Comment by mdkeil** [comment]
Hat schon wer Erfahrungen damit, wenn die Meter Daten als raw-sml vorliegen? --werden so auch versendet, topic ist ebenfalls bekannt.

**Comment by micw** [comment]
Hallo Johannes,

sehr schönes Projekt und vielen Dank für den Link auf mein Blog. Ich hatte auch schon darüber nachgedacht, den Pulse so zu ersetzen. Leider hat mir bisher die Zeit gefehlt, die MQTT-Daten abzugreifen und zu analysieren. Außerdem läuft meine Alternative Lösung (Zähler per IR lessen und die Daten per IR 1:1 wieder ausgeben) sehr stabil. Das würde ja auch mit Deiner Lesehardware funktionieren.

Hättest Du vielleicht Lust, die MQTT-Kommunikation ein wenig genauer zu dokumentieren? Also wann welche Paket auf welcher Topic kommt und wie die Pakete mit den SML-Daten aussehen?

Viele Grüße,
Michael.

**Comment by jsphuebner** [comment]
Hallo Michael,

danke für den Blogbeitrag, ohne den wärs schwierig gewesen ins Thema einzusteigen
Die Variante hatte ich mir auch überlegt, wenn ich das direkte Senden nicht hinbekommen hätte.

Kann noch etwas zur Kommunikation schreiben, sie ist im Grunde recht simpel. Im Echtzeit-Takt gehen die OBIS Daten raus und alle paar Minuten die Metrics. Einmal am Start das "Hello"

**Comment by jsphuebner** [comment]
> Hat schon wer Erfahrungen damit, wenn die Meter Daten als raw-sml vorliegen? --werden so auch versendet, topic ist ebenfalls bekannt.

Das Topic ist bekannt? Wie lautet es?

**Comment by mdkeil** [comment]
`$aws/rules/ingest_tibber_bridge_data/tibber-bridge/<UID>/publish/TFD01/<EUI>/sml`

**Comment by jsphuebner** [comment]
Danke :+1: Ich habe in der tibber readme jetzt ein paar Infos übers Protokoll reingeschrieben

**Comment by mdkeil** [comment]
Kein Problem.. die topics sind aber glaube case sensitive, sprich das SML am Ende muss komplett klein geschrieben werden.

**Comment by Holger237237** [comment]
Hallo Zusammen,
Bis vor ein paar Tagen hatte ich ein SML-Lesekopf der mit zuverlässig Daten der Tasmota über MQTT an meinen Openhab-Server lieferte. 
Jetzt habe ich stattdessen einen Pulse, der zwar Tibber stabil beliefert, jedoch fällt die Datenlieferung an Openhab immer wieder aus. 
Das Projekt hier könnte da wohl eine Lösung liefern. Ich habe es aber noch nicht ganz verstanden. Könnte ich meine alte stabile Lösung wieder im Betrieb nehmen und meine MQTT-Daten zu Tibber senden? Wo muss ich dann was ergänzen? Im MQTT-Server? Oder im Tasmota?
Danke Holger

**Comment by jsphuebner** [comment]
Tasmota und openHAB sagt mir nur bedingt was, das Projekt hier ist ja quasi "bare metal". Aber genau das ist das Ziel, den Pulse durch einen eigenen Lesekopf zu ersetzen. Du brauchst nur die Rohdaten vom Zähler auf einem beliebigen MQTT topic.

**Comment by Holger237237** [comment]
Die Rohdaten liegen mir ja als MQTT topic vor. Der Hardware Teil wäre also
erledigt. Nur wie ich dies dann zu Tibber bekomme, habe ich nicht
verstanden.


Danke

Johannes Huebner ***@***.***> schrieb am So., 29. Okt. 2023,
08:11:

**Comment by mdkeil** [comment]
Wenn du die Daten nicht in kurzen Intervallen für Steuerungsaufgaben benötigst, reicht es auch, wenn du dir die Zählerdaten von der Bridge ausliest.. 5-10s Intervall sollte da problemlos möglich sein.

**Comment by micw** [comment]
@jsphuebner Ist es egal, ob man TFD01/<eui>/obis_stream oder TFD01/<eui>/SML sendet? Wie sieht so ein "obis_stream" denn aus? Und ist "binary SML format" tatsächlich ein Binär-Blob? Sind das genau die Daten, die man auch im Pulse angezeigt bekommt?

Obis_Stream finde ich ja sehr interessant, da Tibber noch nicht jeden Zähler unterstützt. Wenn man selbst die Daten als OBIS bereitstellen kann, könnte man darüber generisch jeden Zähler unterstützen. Ich könnte mir vorstellen, das als weitere Datensenke in https://github.com/micw/homedatabroker zu implementieren.

Ich habe mich übrigens für eine reine Hardware-Lösung entschieden. Damit funktioniert Tibber unabhängig von meiner eigenenInfrastruktur: https://github.com/micw/ir-lesekopf

**Comment by Holger237237** [comment]
Dieses interval wäre mehr als ausreichend. Aktuell nutze ich nur 30s

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by mdkeil** [comment]
@micw 

unter /sml wird genau das erwartet, was man auf der Bridge auch unter Daten abrufen kann. Der Stream wird auch so Richtung Tibber verschickt..

**Comment by micw** [comment]
> Dieses interval wäre mehr als ausreichend. Aktuell nutze ich nur 30s

Das kannst du mir meinem Tool ohne großen Aufwand machen. Tibber Http als source, mqtt als Output, starten, fertig ;-)

**Comment by mdkeil** [comment]
> Wie sieht so ein "obis_stream" denn aus?

Da würde ich mich anschließen.. wollte eigentlich den raw-sml stream schicken.. das was ich aber auslese ist 24 byte kleiner-- keine Ahnung, inwiefern der übermittelte sml-stream noch Zusatzdaten enthält.
Ich würde dann eher mal versuchen einen entsprechenden obis-stream aus meinen Daten zu erzeugen.. ist glaube einfacher.

**Comment by micw** [comment]
`tibber obis_stream` ist genug Futter für Google, hab das hier gefunden: https://openinverter.org/forum/viewtopic.php?p=60335&sid=c1e87bdf5e381e16a5ecfe574e2557dd#p60335

> by [johu](https://openinverter.org/forum/memberlist.php?mode=viewprofile&u=2) » Thu Aug 17, 2023 1:31 pm
> 
> I have finally cracked it. It seems tibber now supports my meter so I was able to intercept some actual data packets.
> 
> The data is sent in topic `$aws/rules/ingest_tibber_bridge_data/tibber-bridge/xxx/publish/TJH01/yyy/obis_stream`
> xxx is the long uid and yyy the eui (see metric)
> 
> The contents is just the raw data read from the meter with one newline removed
> 
> ```
> /EBZ5DD3BZ06ETA_107
> 1-0:0.0.0*255(1EBZzzzz)
> 1-0:96.1.0*255(1EBZzzzz)
> 1-0:1.8.0*255(009094.27916715*kWh)
> 1-0:16.7.0*255(000169.25*W)
> 1-0:36.7.0*255(000133.51*W)
> 1-0:56.7.0*255(000035.74*W)
> 1-0:76.7.0*255(000000.00*W)
> 1-0:32.7.0*255(233.4*V)
> 1-0:52.7.0*255(234.5*V)
> 1-0:72.7.0*255(234.6*V)
> 1-0:96.5.0*255(001C0104)
> 0-0:96.8.0*255(08F0EAB3)
> !
> ```
> 
> I am currently brushing up the code then will publish the python script. You will need to extract the longer tibber uid that is found in the parameter "mqtt_topic", the bridge id that you can obtain by running the "version" command on the console, and the "EUI" of the actual IR device which is found on the "Nodes" tab.
> Then you also need to extract the three certificates for the MQTT connection into PEM files.
>

**Comment by mdkeil** [comment]
Das hatte ich auch gefunden, bringt mich aber nur bedingt weiter.. das scheint auch kein Zweirichtungszähler zu sein.

**Comment by micw** [comment]
Der hat ja auch nur andere OBIS-Codes

Edit: https://de.wikipedia.org/wiki/OBIS-Kennzahlen

**Comment by mdkeil** [comment]
stimmt natürlich.. ich brauche es zum Glück aber nicht mehr.. habe es mit dem sml-stream via node red hinbekommen.. Werte werte angenommen und angezeigt.. werde mich morgen dann nochmal an die Metriken machen. Wenn alles fertig ist, werde ich das Ergebnis zur Verfügung stellen.

**Comment by TheCutter** [comment]
> stimmt natürlich.. ich brauche es zum Glück aber nicht mehr.. habe es mit dem sml-stream via node red hinbekommen.. Werte werte angenommen und angezeigt.. werde mich morgen dann nochmal an die Metriken machen. Wenn alles fertig ist, werde ich das Ergebnis zur Verfügung stellen.

Welches Plugin für Node-Red hast du genutzt um SML Binary lesen zu können? Das Smartmeter Plugin funktioniert für unseren Anwendungsfall ja nicht...

**Comment by mdkeil** [comment]
Ich nutze auf meinem RPI ser2net, um den entsprechenden Port auszulesen und die Ausgabe auf einen lokalen tcp-port zu binden. In Node Red nutze ich dann den "tcp in"-Node, verbinde mich auf den host:port mit Ausgabe in einem Buffer. Pro raw-sml-stream ergibt das 30separate messages (29x 8byte Buffer, 1x 4byte Buffer), die ich dann via Join-Node (manuell, binärer Buffer, verbunden mit buffer [keine Eintragung] zu einer Nachricht (260byte Buffer) verbinde. Die 4byte Buffer Nachricht bekommt msg.complete. Die 2 Metriken und die Event-Message habe ich ebenfalls umgesetzt.. Muss ich heute Abend dann alles mal Testen.

**Comment by mdkeil** [comment]
Test erfolgreich-- läuft seit über 2h durch.
Edit: 31.10. // über 24h problemlos 

![tibber_send_node_red](https://github.com/jsphuebner/esp-egycounter/assets/8126652/7cb5aa33-cf1a-4c0a-86db-5de3c30f07d4)


[tibbersend_Node-RED.json](https://github.com/jsphuebner/esp-egycounter/files/13208749/tibbersend_Node-RED.json)

**Comment by TheCutter** [comment]
Danke! Ich habe ein Wemos D1 Mini mit einer modifizierten Version von https://github.com/mruettgers/SMLReader laufen, der mir die Daten als Raw Binary und geparstes SML in MQTT published. So kann ich die Daten direkt selber verwenden und die Binärdaten direkt an Tibber weiterleiten. So zumindest mein Plan.

Habt ihr euch schon Gedanken gemacht zu den Metriken in Bezug auf Versionsnummern und OTA Updates? Der Pulse kann ja auch Kommandos empfangen von Tibber. Da wird die Selbstbaulösung ja nicht drauf reagieren.

**Comment by jsphuebner** [comment]
> Habt ihr euch schon Gedanken gemacht zu den Metriken in Bezug auf Versionsnummern und OTA Updates? Der Pulse kann ja auch Kommandos empfangen von Tibber. Da wird die Selbstbaulösung ja nicht drauf reagieren.

Nein, noch nicht. Eine Idee wäre mal alles mitzuloggen, was von Tibber so reinkommt. Momentan wird das nur auf den lokalen MQTT weitergeleitet und niemand liest es.
Eventuell müsste man dann die Versionsnummern in den Metrics anpassen als Reaktion?

**Comment by TheCutter** [comment]
Ah okay, ja mitloggen würde ich es auch und dann mal schauen. Die Versionsnummer ändern wäre vermutlich irgendwann sinnvoll. :)

**Comment by mdkeil** [comment]
Ähnliches habe ich auch vor.. wenn sich von der reinen Struktur der Metriken nichts ändert, sollte es auch kein großes Problem sein eine automatische Änderungen der Metriken zu erzeugen.. was ich am Ende des Tages aber vermeiden will, dass ich hier und da wieder die Bridge anschließen muss, um diese zu aktualisieren um dann schlussendlich die neuen Metriken zu erhalten..

**Comment by mdkeil** [comment]
> Eine Idee wäre mal alles mitzuloggen, was von Tibber so reinkommt. Momentan wird das nur auf den lokalen MQTT weitergeleitet und niemand liest es.

Sind denn die topics ggfs. schon bekannt?

**Comment by TheCutter** [comment]
> Sind denn die topics ggfs. schon bekannt?

`tibber-bridge/<UID>/receive`

ist das Topic für eingehende Nachrichten.

**Comment by irqnet** [comment]
Hat jemand ne grobe Idee in welche Richtung ich suchen muss, warum ich hier laufend CONNECT Meldungen bekomme?

Initial hatte das Ganze einmal funktioniert, aber seitdem der Pulse mit dem IR in Betrieb ist und in der App Werte anzeigt. stehe ich via MQTT scheinbar nichts mehr, oder der Connect wird abgewiesen?!

![image](https://github.com/jsphuebner/esp-egycounter/assets/32534199/faf9c275-077b-40c0-b9bf-e9ffa208c84e)

**Comment by mdkeil** [comment]
> > Habt ihr euch schon Gedanken gemacht zu den Metriken in Bezug auf Versionsnummern und OTA Updates? Der Pulse kann ja auch Kommandos empfangen von Tibber. Da wird die Selbstbaulösung ja nicht drauf reagieren.
> 
> Nein, noch nicht. Eine Idee wäre mal alles mitzuloggen, was von Tibber so reinkommt. Momentan wird das nur auf den lokalen MQTT weitergeleitet und niemand liest es. Eventuell müsste man dann die Versionsnummern in den Metrics anpassen als Reaktion?

kurze Rückmeldung.. mein [node-red-flow](https://github.com/jsphuebner/esp-egycounter/issues/2#issuecomment-1785839199) läuft seit Oktober '23 durch, ohne eine Anpassung der Metriken.. habe bis dato glaube auch auf dem tibber-bridge/<UID>/receive - topic noch nichts erhalten oder ich habe es übersehen ;)

**Comment by mdkeil** [comment]
@jsphuebner 
@micw 

Eine kurze Frage.. Habe für einen anderen User meinen flow angepasst, da er einen nicht tibber-kompatiblen Zähler nutzt. folgender obis_stream wird ausgelsen und auf das entsprechende topic geschickt.

```
/LOG5LK13BE803039
1-0:96.1.0*255(001LOG0065477994)
1-0:1.8.0*255(010344.4321*kWh)
1-0:2.8.0*255(000000.0000*kWh)
1-0:16.7.0*255(000426*W)
1-0:32.7.0*255(239.0*V)
1-0:52.7.0*255(238.8*V)
1-0:72.7.0*255(239.3*V)
1-0:31.7.0*255(000.17*A)
1-0:51.7.0*255(001.64*A)
1-0:71.7.0*255(000.32*A)
1-0:81.7.1*255(117*deg)
1-0:81.7.2*255(241*deg)
1-0:81.7.4*255(030*deg)
1-0:81.7.15*255(036*deg)
1-0:81.7.26*255(014*deg)
1-0:14.7.0*255(49.9*Hz)
1-0:1.8.0*96(00006.6*kWh)
1-0:1.8.0*97(00047.1*kWh)
1-0:1.8.0*98(00144.3*kWh)
1-0:1.8.0*99(00865.4*kWh)
1-0:1.8.0*100(10344.4*kWh)
1-0:0.2.0*255(ver.03,432F,20170504)
1-0:96.90.2*255(0F66)
1-0:97.97.0*255(00000000)
!
```
In der App laufen aber keine Daten ein, die Metriken schon. Kann es ggfs sein, dass "gewisse" Zählerdaten abgeglichen werden mit der Kompatibilitätsdatenbank? Wenn ja, müsste man diese ja vorab ersetzen und dadurch einen anderen Zählertyp vortäuschen?! Oder ob der obis_stream zu "komplex" ist? Wie gesagt, alles reine Mutmaßung.

**Comment by kalaws** [comment]
> @jsphuebner @micw
> 
> Eine kurze Frage.. Habe für einen anderen User meinen flow angepasst, da er einen nicht tibber-kompatiblen Zähler nutzt. folgender obis_stream wird ausgelsen und auf das entsprechende topic geschickt.
> 
> ```
> /LOG5LK13BE803039
> 1-0:96.1.0*255(001LOG0065477994)
> 1-0:1.8.0*255(010344.4321*kWh)
> 1-0:2.8.0*255(000000.0000*kWh)
> 1-0:16.7.0*255(000426*W)
> 1-0:32.7.0*255(239.0*V)
> 1-0:52.7.0*255(238.8*V)
> 1-0:72.7.0*255(239.3*V)
> 1-0:31.7.0*255(000.17*A)
> 1-0:51.7.0*255(001.64*A)
> 1-0:71.7.0*255(000.32*A)
> 1-0:81.7.1*255(117*deg)
> 1-0:81.7.2*255(241*deg)
> 1-0:81.7.4*255(030*deg)
> 1-0:81.7.15*255(036*deg)
> 1-0:81.7.26*255(014*deg)
> 1-0:14.7.0*255(49.9*Hz)
> 1-0:1.8.0*96(00006.6*kWh)
> 1-0:1.8.0*97(00047.1*kWh)
> 1-0:1.8.0*98(00144.3*kWh)
> 1-0:1.8.0*99(00865.4*kWh)
> 1-0:1.8.0*100(10344.4*kWh)
> 1-0:0.2.0*255(ver.03,432F,20170504)
> 1-0:96.90.2*255(0F66)
> 1-0:97.97.0*255(00000000)
> !
> ```
> 
> In der App laufen aber keine Daten ein, die Metriken schon. Kann es ggfs sein, dass "gewisse" Zählerdaten abgeglichen werden mit der Kompatibilitätsdatenbank? Wenn ja, müsste man diese ja vorab ersetzen und dadurch einen anderen Zählertyp vortäuschen?! Oder ob der obis_stream zu "komplex" ist? Wie gesagt, alles reine Mutmaßung.

Is your solution working without you having bought a Pulse first? Think it would work in Sweden? We don’t have Tibber Bridge I think. 

Update: Actually, what I would ideally like to do is to publish solar production data as I have an inverter that’s not supported by Tibber.

**Comment by mdkeil** [comment]
you need to buy a pulse first.. I think It should work in Sweden too.. but I think, the mqtt-topics are different and you have to do some trial and error.. its not that simple I think.. for your solar inverter I don't think, that there is a simple mqtt-solution.

**Comment by XL-Maggus** [comment]
Hi, 

auch wenn ich das alte Issue wieder rauskramen muss... aber würde das ganze auch mit einem Hichi Wlan Reader funktionieren?

VG

Markus

**Comment by mdkeil** [comment]
Denke schon.. der kann doch auch direkt via mqtt publishen oder?

**Comment by XL-Maggus** [comment]
kann er ... ich hab nur immer noch nich die genaue sml struktur gefunden, die tibber dann für live daten akzeptiert. 
dann könnte ich das tasmota SML script anpassen.

**Comment by mdkeil** [comment]
so einfach ist es nicht.. nur die live daten senden reicht ja nicht.. was und wie findest du hier im entsprechenden python-script oder in meiner node-red Umsetzung--  tibber akzeptiert entweder einen sml oder obis-stream.

**Comment by XL-Maggus** [comment]
wenn es so einfach wäre ... aktuell bekomm ich weder das eine, noch das andere ans laufen, damit mir die realdaten angezeigt werden. Ich hol die daten mit nem pi über den tasmota ab und streame /dev/ttyUSB0 live über socat an einen tcp port ...

**Comment by mdkeil** [comment]
https://github.com/jsphuebner/esp-egycounter/issues/2#issuecomment-1785839199

dann schau hier mal rüber-- habe alles bestmöglich beschrieben-- deine Daten "docken" links unten an.. settings und topics wie beschrieben anpassen.

..oder halt hier schauen, wenn es python sein soll. 
https://github.com/jsphuebner/esp-egycounter/blob/main/battery-control/tibbersend.py

**Comment by XL-Maggus** [comment]
kannst du mir mal deine ser2net config geben?

**Comment by mdkeil** [comment]
`3000:raw:600:/dev/ttyS1:19200 8DATABITS NONE 1STOPBIT max-connections=2`

**Comment by XL-Maggus** [comment]
hmmm also mit dem sml stream bekomm ich keine live anzeige hin ... funktioniert die bei dir?

**Comment by mdkeil** [comment]
da ich nicht weiß, was du bis dato wie gemacht hast, kann ich nicht helfen.. zumal ich auf meinem mqtt Server eine entsprechende Bridge eingerichtet habe für tibber.. alles nicht trivial

**Comment by XL-Maggus** [comment]
hi,

ich hab sml per stream und deine nodered template verwendet. verbunden ist alles, der sml stream ist valide... aber daten zeigt tibber nicht an...

EDIT: 

es funktioniert ... hab bei den topics das $aws/ rausgenommen :D

**Comment by mdkeil** [comment]
supi, warum auch das dein Problem war.. baust du direkt via node red die Verbindung zum tibber mqqt Server auf oder so wie ich über eine eingerichtete Bridge bei einem eigenem separaten mqtt Server?

**Comment by XL-Maggus** [comment]
hi, 

ich hatte das direkt in nodered drin... 

aber seit gestern funktioniert es nicht mehr, weder nr noch das script akzeptieren meine sml daten... langsam verzweifel ich echt an tibber 😂

**Comment by XL-Maggus** [comment]
hi, 
leider kenn ich mich in nodered nicht aus, aber irgendwie scheint er das ganze zwischen tcp-stream und TFD01-sml nicht zu fressen ... 

wenn ich den tcp stream direkt an einen lokalen mqtt schicke ... kommen sekündlich die rohdaten vom uart an.

<img width="1422" alt="Image" src="https://github.com/user-attachments/assets/4d0a5335-926d-451a-968d-0eceffd6666d" />

**Comment by ggrote** [comment]
Laut des Screenshots sendest du die Rohdaten ja 3x an den TFD01-sml Endpunkt. Das ist natürlich zu viel. Das solltest du rausnehmen. Die anderen Nodes oben drüber aus dem NodeRed Template hast du aber auch laufen? Tibber ist da sehr empfindlich. Settings für TJH01 hast auch alle angepasst? $aws muss als Topic schon drin stehen. Da sind viele Sachen irgendwie komisch...

**Comment by mdkeil** [comment]
Kurz zur Erklärung-- der Anteil hinter "sml-fromTCP" ist notwendig, um den raw-stream so zu "verpacken", wie tibber ihn erwartet. Die Mqtt-Verbindung zwischen dem mqqt-node in node red und dem tibber-mqtt steht aber? Und wie bereits angemerkt solltest Du die Daten nicht 3 mal an den TFD01-sml node schicken.

**Comment by XL-Maggus** [comment]
ich sende die daten natürlich nicht an tibber, sondern erstmal an meinen eigenen mqtt ... unter verschiedene topics damit ich mitlesen kann. 

ich hab die topics angepasst und das funktioniert auch der stream der durch setStart ... pubCnt ... TFD01-sml geht sollte auf meinem mqtt unter sml aufschlagen, da kommt nur leider gar nix an... 

leite ich direkt vom tcp port an ein mqtt topic weiter kommen die roh stream daten unter rawstream

<img width="1289" alt="Image" src="https://github.com/user-attachments/assets/ce790755-a42a-4c3e-9c96-4715bc3221f5" />

 an.

**Comment by XL-Maggus** [comment]
ja, die verbindung steht, aber mein mqtt kann halt nichts durchleiten, wenn bei $aws/rules/ingest_tibber_bridge_data/tibber-bridge/XXXX/publish/TFD01/YYYYY/sml nichts aus dem node-red rausfällt.

**Comment by ggrote** [comment]
Ah okay. Ist bei dir im Flow wo die Tibber Nodes sind denn "tibber.publish" auf true gesetzt? Wenn du ganz rechts oben in Node Red auf den kleinen Pfeil klickst und dann auf Kontextdaten. Bei Flow dann rechts einmal auf das Reload Symbol klicken. Dann müsste da etwas erscheinen wo "publish: true" dabei stehen sollte.

**Comment by XL-Maggus** [comment]
das ist gesetzt ja ... kann es sein, dass mein efr einfach eine art grösseren datensatz als erwartet schickt?
wenn ich den roh stream mit sml2mqtt auswerte und mir das ganze als obis anzeigen lasse, funktioniert das... muss ich denn die rohdaten noch irgendwie anpasse, aktuell lese ich die mit einem esp32 aus und leite den uart einfach roh durch.

**Comment by mdkeil** [comment]
Sitzt denn dein Lesekopf korrekt!? (und liefert korrekte Daten?).. denn wenn hinten nichts rauskommt, lauft vorne was schief.. und da es bei dir schon funktioniert hat, liegt es nicht an den nodes.

**Comment by ggrote** [comment]
> das ist gesetzt ja ... kann es sein, dass mein efr einfach eine art grösseren datensatz als erwartet schickt? wenn ich den roh stream mit sml2mqtt auswerte und mir das ganze als obis anzeigen lasse, funktioniert das... muss ich denn die rohdaten noch irgendwie anpasse, aktuell lese ich die mit einem esp32 aus und leite den uart einfach roh durch.

Habe ich auch schon überlegt. Ich nutze den TCP Stream nicht, von daher kann ich dir da nicht helfen. Ich sende die Daten direkt per MQTT über https://github.com/mruettgers/SMLReader an meinen MQTT und leite sie über Node Red weiter an AWS/Tibber. Funktioniert super!

**Comment by jsphuebner** [comment]
Wollte auch nur kurz bestätigen, dass mein Skript weiterhin unverändert läuft!
Es hat allerdings auch einen Bug, die metrics werden gleich mehrfach gesendet. Scheint aber nicht zu stören. Sämtliche Versuche es zu beheben, haben dazu geführt, dass Tibber keine Daten mehr angezeigt hat

**Comment by mdkeil** [comment]
> kann es sein, dass mein efr einfach eine art grösseren datensatz als erwartet schickt?

Will ich nicht ausschließen.. ich habe halt im Vorfeld geschaut, was der tibber puls als raw-stream über die Bridge versendet und habe dann verglichen, was mein Hitchi-Lesekopf liefert bzw. was auf dem TCP-Port dann in Node-Red ankommt. Inhaltlich passe ich ja nichts an.. ich packe es nur in entsprechende buffer.. vielleicht kann ich heute Abend mal schicken, in welchem Format bei mir die Rohtdaten kommen und was ich an Tibber schicke..

**Comment by XL-Maggus** [comment]
> Sitzt denn dein Lesekopf korrekt!? (und liefert korrekte Daten?).. denn wenn hinten nichts rauskommt, lauft vorne was schief.. und da es bei dir schon funktioniert hat, liegt es nicht an den nodes.

ja, der lesekopf funktioniert und liefert alle daten ... ich hab hier aktuell 2 lese/schreib köpfe im einsatz, da ich für eine nulleinspeisung zwingend auf realtime daten angewiesen bin und tibber ist mir da zu langsam. Das Tasmota das hier mitläuft, meldet keinerlei fehler und permanent valide daten.

Ich glaube, dass es schon funktioniert hat, war ein trugschluss, da ich die bridge und den pulse nur per software vom internet getrennt habe, diese scheint aber die aktive mqtt connection nicht getrennt zu haben...

**Comment by XL-Maggus** [comment]
> Habe ich auch schon überlegt. Ich nutze den TCP Stream nicht, von daher kann ich dir da nicht helfen. Ich sende die Daten direkt per MQTT über https://github.com/mruettgers/SMLReader an meinen MQTT und leite sie über Node Red weiter an AWS/Tibber. Funktioniert super!

spukt das bvei dir daten in so einer form aus? kannst du mir mal diesen flow schicken?

<img width="296" alt="Image" src="https://github.com/user-attachments/assets/fc3c5203-f58d-4780-9428-73ffbff4e73b" />

**Comment by ggrote** [comment]
> Ich glaube, dass es schon funktioniert hat, war ein trugschluss, da ich die bridge und den pulse nur per software vom internet getrennt habe, diese scheint aber die aktive mqtt connection nicht getrennt zu haben...

Warte mal, das könnte das Problem sein! Du darfst den Bridge und den Pulse nicht bei Tibber aus der App entfernen! Sonst geht Tibber ja davon aus, dass du gar keine Daten mehr sendest. Und tust es dann aber trotzdem.

Das ist mein Flow für die Rohdaten. :D
<img width="528" alt="Image" src="https://github.com/user-attachments/assets/2fdbdc89-70d8-47e1-af3d-e4ab0fd1c3f2" />

**Comment by XL-Maggus** [comment]
> > Ich glaube, dass es schon funktioniert hat, war ein trugschluss, da ich die bridge und den pulse nur per software vom internet getrennt habe, diese scheint aber die aktive mqtt connection nicht getrennt zu haben...
> 
> Warte mal, das könnte das Problem sein! Du darfst den Bridge und den Pulse nicht bei Tibber aus der App entfernen! Sonst geht Tibber ja davon aus, dass du gar keine Daten mehr sendest. Und tust es dann aber trotzdem.
> 


natürlich entferne ich nicht die bridge und den pulse aus tibber ... ich trennen sie eigentlich nur vom netz.


ich versuch das mal mit deiner methode ... muss dann aber nicht /sml sondern /obis_stream sein oder?

**Comment by mdkeil** [comment]
> muss dann aber nicht /sml sondern /obis_stream sein oder?

Das ist davon abhängt, was dein Zähler liefert

**Comment by mdkeil** [comment]
Habe nochmal geschaut.. also auf meinem TCP Port kommen die Daten in Datenblöcken buffer [xx xx xx xx xx xx xx xx] an; 8 Buffer bilden den gesamten sml-Stream, welcher 260bytes groß ist bei mir.. Es werden also 260bytes in einem buffer gejoint und übertragen.. Heute Abend poste ich es nochmal ordentlich, wenn ich Zugriff habe..

**Comment by XL-Maggus** [comment]
ok ... ich hab grad mal geschaut, was mir die bridge bei einem sml raw dump ausgibt, hier das stat der datei: 
Datei: dump
Größe: 596       	Blöcke: 8          EA Block: 4096   reguläre Datei
Gerät: 254/1	Inode: 14103       Verknüpfungen: 1

angehängt hab ich einmal die datei die ich mir direkt vom pulse geholt und durch pysmlparse gejagt habe.

[dump.txt](https://github.com/user-attachments/files/18767233/dump.txt)

**Comment by mdkeil** [comment]
> spukt das bvei dir daten in so einer form aus? kannst du mir mal diesen flow schicken?

aber wenn du doch eh sml2mqqt nutzt, dann hole dir doch die Daten direkt in node-red ab und setze es um wie @ggrote 

https://github.com/jsphuebner/esp-egycounter/issues/2#issuecomment-2653289586

**Comment by ggrote** [comment]
> spukt das bvei dir daten in so einer form aus? kannst du mir mal diesen flow schicken?

Ich habe es angepasst, damit ich die Daten auch im RAW Format erhalte.
Habe die Änderungen mal hier hochgeladen:
https://github.com/ggrote/SMLReader

**Comment by XL-Maggus** [comment]
hi, danke ich versuch es mal ... muss das ganze aber nach esphome portieren :D

**Comment by XL-Maggus** [comment]
hab da nun rumgetestet und rumgebastelt ... alles nicht funktioniert mit tcp streams, auf blöd hab ich es nun mal mit direkt serial aufgebaut, mein zähler schickt 592B grosse paketet mit einer pause von ca 250ms ...  so funktioniert es nun bei mir.

<img width="1029" alt="Image" src="https://github.com/user-attachments/assets/cae4550f-c653-4d43-9259-c22cfcb37b3a" />

**Comment by mdkeil** [comment]
Alle Wege führen nach Rom! ;)

**Comment by mdkeil** [comment]
Bei mir wurde gestern der Zähler auf iMSys ausgetauscht.. direkt abends war der neue Zähler bereits bei Tibber hinterlegt mit der Information, das mein Netzbetreiber(EWE) ab dem 01.04. 00:00 die Werte an  Tibber übermittelt. 
Da mir für die Freischaltung des neuen Zählers noch die PIN fehlt, ich aber bis Ende des Monats keine Lücke haben wollte, was die stündliche Abrechnung angeht, habe ich mir schnell aus meinen vorhandenen Smart-Meter-Daten einen minimalen (akzeptierten) obis-Stream zusammengesetzt und diesen übermittelt.. funktioniert!

//Zählernummer sanitarisiert ;)
```
/LGZ00xxxxxxxx
1-0:0.0.0*255(1LGZzzzz)
1-0:96.1.0*255(1LGZzzzz)
1-0:1.8.0*255(000000.2790*kWh)
1-0:16.7.0*255(-000002*W)
!
```

**Comment by XL-Maggus** [comment]
ich würd ja nun sagen, brute force den zähler ... kannst mich gerne mal per email kontaktieren :-)
